core/mobile-manager/src/backend/iOSHost.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { ClientRequestContext } from "@bentley/bentleyjs-core";
+import { IModelHostConfiguration, IpcHostOptions } from "@bentley/imodeljs-backend";
+import { IpcWebSocketBackend } from "@bentley/imodeljs-common";
+import { BatteryState, Orientation } from "../common/MobileAppProps";
+import { DownloadTask, MobileCompletionCallback, MobileDevice, MobileDeviceAuthSettings, MobileHost, MobileProgressCallback } from "./MobileHost";
+
+type DeviceEvents = "memoryWarning" | "orientationChanged" | "enterForeground" | "enterBackground" | "willTerminate";
+
+/** @beta */
+export class IOSDevice extends MobileDevice {
+  public emit(eventName: DeviceEvents, ...args: any[]) {
+    switch (eventName) {
+      case "memoryWarning":
+        MobileHost.onMemoryWarning.raiseEvent(...args); break;
+      case "orientationChanged":
+        MobileHost.onOrientationChanged.raiseEvent(...args); break;
+      case "enterForeground":
+        MobileHost.onEnterForeground.raiseEvent(...args); break;
+      case "enterBackground":
+        MobileHost.onEnterBackground.raiseEvent(...args); break;
+      case "willTerminate":
+        MobileHost.onWillTerminate.raiseEvent(...args); break;
+    }
+  }
+
+  public getOrientation(): Orientation { throw new Error("No native implementation registered."); }

```

Comment:
@swbsi Don't all these need implementations before we can complete this PR?

Reply:
> @swbsi Don't all these need implementations before we can complete this PR?

They are implemented on the native side. We find this object (using the global name) and set all those properties with napi::functions.

Reply:
maybe we can just leave this IOSDevice abstract instead of having these "dummy" implementations? You'll have to do `as any` where you call the ctor.

Reply:
and put a comment that they're actually implemented in native code.

Reply:
Actually I just realized that you won't be able to derive from iosdevice if we stomp on its functions. I will push a NativeBridge class (abstract) that behaves as you describe and point the native side there.

Reply:
and the name on the global object shouldn't include "RpcImpl", if you can change that.

Reply:
> and the name on the global object shouldn't include "RpcImpl", if you can change that.

That would be a native side change.

Reply:
but why not change the native side too in [this pr](https://bentleycs.visualstudio.com/iModelTechnologies/_git/imodel02/pullrequest/141613)?

Reply:
done